dessia_common/workflow/core.py

```diff
@@ -2186,6 +2186,67 @@ def _method_jsonschemas(self):
         jsonschemas['run_again']['classes'] = ["dessia_common.workflow.WorkflowRun"]
         return jsonschemas
 
+    def to_script(self):
+        """
+        Computes a script representing the workflowrun.
+        """
+        # TODO : don't take into account yet TypedVariable.
+        # TODO : take into account the type of inputs (type_)
+        workflow_script = self.workflow.to_script()
+        workflow_script_import = self.workflow._to_script().imports
+        input_str = ""
+        default_value = ""
+        add_import = ""
+        var_1 = "block_{}.inputs[{}]"
+        for j, block in enumerate(self.workflow.blocks):
+            for i, input_ in enumerate(block.inputs):
+                if not input_.has_default_value and not \
+                        var_1.format(j, i) in workflow_script:
+                    input_str += f"    workflow.input_index(" \
+                                 f"{var_1.format(j, i)}):" \
+                                 f" value_{str(j) + '_' + str(i)},\n"
+                    default_value_ = f"\nvalue_{j}_{i} = 0"
+                    if not isinstance(input_, (str, float, int)):
+                        try:
+                            module_ = input_._get_to_script_elements().get_import_dict()
+                            key, value = list(module_.items())[0]
+                            default_value_ = f"\nvalue_{j}_{i} = {value[0]}()"
+                        except:
+                            pass
+                        if not any(value[0] in lst for lst in (add_import, workflow_script_import)):
+                            add_import += f"from {key} import {value[0]}\n"
+                        default_value += default_value_
+        for k, nbv in enumerate(self.workflow.nonblock_variables):
+            if not nbv.has_default_value:
+                if nbv.type_ == str:
+                    default_value += f"\nvalue_nbv_{k} = 'str'"
+                if nbv.type_ == int:
+                    default_value += f"\nvalue_nbv_{k} = 'int'"
+                input_str += f"    workflow.input_index(" \
```

Comment:
Instead of writing input_index into the string, we could compute the right value for the index beforehand and write `f"{{global_index} : value_nbv_{k}}`. 

You can also get rid of the `str()` in fstrings

Be aware of the style conventions in computed code (spaces around " : ", space only after ", ",...)